tests/lib/prepare.sh

```diff
@@ -214,6 +214,11 @@ update_core_snap_for_classic_reexec() {
     done
 }
 
+flush_changes() {
+    sleep 1
```

Comment:
Why the initial 'sleep 1'? Maybe a comment why it's neccessary, otherwise this looks really good to me!

Reply:
To avoid races where a task is being queued but not yet in `snap changes`.

Reply:
I will add a comment when I get the CI to run again. "retry" seems to have broken it.

Reply:
Done